crates/uv-git/src/git.rs

```diff
@@ -364,6 +364,8 @@ impl GitCheckout {
         ProcessBuilder::new("git")
             .arg("clone")
             .arg("--local")
+            // Propagate the blobless clone.
+            .arg("--filter=blob:none")
```

Comment:
If I comment this one out, the test suite doesn't hang 🎉 but there are still a bunch of failures e.g.

```
Snapshot: reinstall_git
Source: crates/uv/tests/pip_sync.rs:2010
────────────────────────────────────────────────────────────────────────────────
Expression: snapshot
────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬───────────────────────────────────────────────────────────────────
    0       │-success: true
    1       │-exit_code: 0
          0 │+success: false
          1 │+exit_code: 2
    2     2 │ ----- stdout -----
    3     3 │ 
    4     4 │ ----- stderr -----
    5       │-Resolved 1 package in [TIME]
    6       │-Downloaded 1 package in [TIME]
    7       │-Installed 1 package in [TIME]
    8       │- + uv-public-pypackage==0.1.0 (from git+https://github.com/astral-test/uv-public-pypackage@b270df1a2fb5d012294e9aaf05e7e0bab1e6a389)
          5 │+error: Failed to download and build `uv-public-pypackage @ git+https://github.com/astral-test/uv-public-pypackage@b270df1a2fb5d012294e9aaf05e7e0bab1e6a389`
          6 │+  Caused by: Git operation failed
          7 │+  Caused by: process didn't exit successfully: `git reset --hard b270df1a2fb5d012294e9aaf05e7e0bab1e6a389` (exit status: 128)
          8 │+--- stderr
          9 │+error: unable to read sha1 file of .gitignore (e69de29bb2d1d6434b8b29ae775ad8c2e48c5391)
         10 │+error: unable to read sha1 file of README.md (0a50718e2c98ce585c9f06d68e853e06cc1c307f)
         11 │+error: unable to read sha1 file of dist/uv_public_pypackage-0.1.0-py3-none-any.whl (30de426777a0e739ccadd993ea1313b95f6386b4)
         12 │+error: unable to read sha1 file of dist/uv_public_pypackage-0.1.0.tar.gz (4d8eb8d9e7558239356266c836da3bfca3aca8f1)
         13 │+error: unable to read sha1 file of pyproject.toml (f971ef95a81ef15fed29090cdbd46e779f78d8e3)
         14 │+error: unable to read sha1 file of src/uv_public_pypackage/__init__.py (3dc1f76bc69e3f559bee6253b24fc93acee9e1f9)
         15 │+fatal: Could not reset index file to revision 'b270df1a2fb5d012294e9aaf05e7e0bab1e6a389'.
```

I wonder if the CI failures are that the "local" server doesn't support filtering?

Reply:
In https://github.com/astral-sh/uv/pull/4047/commits/7140a27e38b5e5575c83cb02d3341ea14d4c3890, allowing filtering on the "local" server makes the warning "filtering not recognized by server, ignoring" go away.

Reply:
However, setting that option locally the test suite still hangs on my machine.

Reply:
If I revert https://github.com/astral-sh/uv/pull/3950/commits/29839a0a92c59c2546db134d14d73c2c12ee2c35 the test suite hangs on two tests my machine, but the rest of the tests pass.

Reply:
Continuing this commentary....

If I upgrade my local git client the tests fail

```
❯ git -v
git version 2.45.2
❯ ct allowed_transitive_git_dependency
    Finished `test` profile [unoptimized + debuginfo] target(s) in 0.14s
    Starting 1 test across 52 binaries (1008 skipped; run ID: f9d63952-2ca1-4121-9989-2d78257fbdc3, nextest profile: default)
        FAIL [   1.171s] uv::pip_compile allowed_transitive_git_dependency
error: test run failed

❯ export PATH="/usr/bin:$PATH"
❯ git -v
git version 2.39.3 (Apple Git-146)
❯ ct allowed_transitive_git_dependency
    Finished `test` profile [unoptimized + debuginfo] target(s) in 0.14s
    Starting 1 test across 52 binaries (1008 skipped; run ID: f9d63952-2ca1-4121-9989-2d78257fbdc3, nextest profile: default)
        PASS [   6.603s] uv::pip_compile allowed_transitive_git_dependency
------------
     Summary [   6.603s] 1 test run: 1 passed, 1008 skipped
```